kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/IfcfgConfigReader.java

```diff
@@ -302,168 +290,97 @@ private Properties parseRedhatConfigFile(File ifcfgFile, String interfaceName) {
-                                             * resulting in multiple entries for
-                                             * the same servers. (Tested on
-                                             * 10-20, 10-10, and Raspberry Pi).
-                                             * Commenting out dns-nameservers in
-                                             * the /etc/network interfaces file
-                                             * allows DNS servers to be picked
-                                             * up by the IfcfgConfigReader and
-                                             * be displayed on the Web UI but
-                                             * the /etc/resolv.conf file will
-                                             * only be updated by Kura.
-                                             */
-                                            if (args.length > 1) {
-                                                for (int i = 1; i < args.length; i++) {
-                                                    kuraProps.setProperty("DNS" + Integer.toString(i), args[i]);
-                                                }
-                                            }
-                                        } else if ("post-up".equals(args[0])) {
-                                            StringBuilder sb = new StringBuilder();
-                                            for (int i = 1; i < args.length; i++) {
-                                                sb.append(args[i]);
-                                                sb.append(' ');
-                                            }
-                                            if (sb.toString().trim().equals("route del default dev " + interfaceName)) {
-                                                kuraProps.setProperty("DEFROUTE", "no");
-                                            }
-                                        }
-                                    }
-                                }
-                                // Debian makes assumptions about lo, handle
-                                // those here
-                                if ("lo".equals(interfaceName) && kuraProps.getProperty("IPADDR") == null
-                                        && kuraProps.getProperty("NETMASK") == null) {
-                                    kuraProps.setProperty("IPADDR", "127.0.0.1");
-                                    kuraProps.setProperty("NETMASK", "255.0.0.0");
-                                }
-                                break;
-                            }
-                        } catch (Exception e) {
-                            logger.warn("Possible malformed configuration file for " + interfaceName, e);
-                        }
-                    }
+    private void setNetInterfaceAddressConfigs(String interfaceName, boolean dhcp, List<NetConfig> netConfigs,
```

Comment:
Is it possible to extract two methods from the branches of the condition to simplify the code?

Reply:
done